lib/encoding/url.toit

```diff
@@ -37,55 +45,52 @@ url-encode_ from -> any:
 
 /**
 Encodes the given $data using URL-encoding, also known as percent encoding.
-The $data must be a string or byte array.  The value returned is a string or
-  a byte array.  It can only be a byte array if the input was a byte array
-  and in this case it can be the identical byte array that was passed in.
+
 The characters 0-9, A-Z, and a-z are unchanged by the encoding, as are the
   characters '-', '_', '.', and '~'.  All other characters are encoded in
   hexadecimal, using the percent sign.  Thus a space character is encoded
   as "%20", and the Unicode snowman (☃) is encoded as "%E2%98%83".
 */
-encode data -> any:
-  if data is string:
-    // If a string is ASCII only then the sizes match.
-    if data.size != (data.size --runes):
-      // Convert to something where do will iterate over UTF-8 bytes.
-      data = data.to-byte-array
-  else if data is not ByteArray:
-    throw "WRONG_OBJECT_TYPE"
+encode data/io.Data -> string:
   return url-encode_ data
 
 /**
 Decodes the given $data using URL-encoding, also known as percent encoding.
 The function is liberal, accepting unencoded characters that should be
   encoded, with the exception of '%'.
-Takes a string or a byte array, and may return a string or a ByteArray.
-  (Both string and ByteArray have a to-string method.)
-Does not check for malformed UTF-8, but calling to-string on the return
-  value will throw on malformed UTF-8.
+
 Plus signs (+) are not decoded to spaces.
 
 # Example
-  (url.decode "foo%20b%C3%A5r").to-string  // Returns "foo bår".
+```
+  url.decode "foo%20b%C3%A5r"  // Returns "foo bår".
+```
+*/
+decode data/string -> string:
+  if not data.contains "%": return data
+  return (decode-binary data).to-string
+
+/**
+Variant of $decode.
+
+Decodes the given $data using URL-encoding and returns the result as a byte array.
+The encoded $data may contain characters that are not valid UTF-8.
 */
-decode data -> any:
-  if data is string:
-    if not data.contains "%": return data
-    if data.size != (data.size --runes): data = data.to-byte-array
-  else if data is ByteArray:
-    if (data.index-of '%') == -1: return data
-  else:
-    throw "WRONG_OBJECT_TYPE"
+decode-binary data/string -> ByteArray:
   count := 0
-  data.do: | c |
-    if c == '%': count++
-  result := ByteArray data.size - count * 2
+  for i := 0; i < data.byte-size; i++:
```

Comment:
Why not use `data.byte-size.repeat` here too?

Reply:
done.